zerver/forms.py

```diff
@@ -139,16 +139,37 @@ def __init__(self, *args: Any, **kwargs: Any) -> None:
             choices=[(lang["code"], lang["name"]) for lang in get_language_list()],
         )
 
+    @override
+    def clean(self) -> None:
+        super().clean()
+        if not self.realm_creation:
+            # The validation below only needs to happen when we are
+            # creating a new realm as part of the registration process.
+            return
```

Comment:
When is `realm_creation` expected to be false? Might be helpful for this comment to hint that.